### PR TITLE
Setup browser testing in CI

### PR DESCRIPTION
**Pull request by webmaster128**
Many of the changes here are required since 3 packages run in parallel now: faucet, cosmwasm, bcp

### CODE REVIEW

File: packages/bcp/package.json [hunk 1]
```diff
@@ -34,7 +34,7 @@
     "build-or-skip": "[ -n \"$SKIP_BUILD\" ] || yarn build",
     "test-node": "node jasmine-testrunner.js",
     "test-firefox": "yarn pack-web && karma start --single-run --browsers Firefox",
-    "test-chrome": "yarn pack-web && karma start --single-run --browsers ChromeHeadlessInsecure",
+    "test-chrome": "yarn pack-web && karma start --single-run --browsers ChromeHeadless",
```
Comment (ethanfrey) [inline packages/bcp/package.json:37]: Always nice to see the word `Insecure` disappear from the codebase :smile:

File: packages/bcp/src/cosmosconnection.spec.ts [hunk 2]
```diff
@@ -41,30 +41,26 @@ function makeRandomAddress(): Address {
...
 
 /**
  * We use a different test account than the faucet, since BCP errors when transactions
  * with multiple messages are found.
  */
 const bob = {
-  mnemonic:
-    "economy stock theory fatal elder harbor betray wasp final emotion task crumble siren bottom lizard educate guess current outdoor pair theory focus wife stone",
-  path: HdPaths.cosmosHub(4),
+  mnemonic: "remain fragile remove stamp quiz bus country dress critic mammal office need",
```
Comment (ethanfrey) [inline packages/bcp/src/cosmosconnection.spec.ts:57]: Why the change here?

Ah, this is to keep in sync with other iov/bcp code rather than in sync with other cosm(wasm)-js code, right?

File: packages/cosmwasm/src/cosmwasmclient.spec.ts [hunk 3]
```diff
@@ -72,9 +74,10 @@ describe("CosmWasmClient", () => {
 
       const height1 = await client.getHeight();
       expect(height1).toBeGreaterThan(0);
-      await sleep(1_000);
+      await sleep(blockTime * 1.3); // tolerate chain being 30% slower than expected
```
Comment (ethanfrey) [inline packages/cosmwasm/src/cosmwasmclient.spec.ts:77]: always good for CI

File: packages/sdk38/src/cosmosclient.searchtx.spec.ts [hunk 4]
```diff
@@ -104,20 +105,22 @@ describe("CosmosClient.searchTx", () => {
           },
         };
         const transactionId = await client.getIdentifier(tx);
-        const heightBeforeThis = await client.getHeight();
         try {
           await client.postTx(tx.value);
         } catch (error) {
           // postTx() throws on execution failures, which is a questionable design. Ignore for now.
```
Comment (ethanfrey) [inline packages/sdk38/src/cosmosclient.searchtx.spec.ts:111]: What alternative would you suggest?

I think it is good to ensure people don't blindly ignore failures (if they don't read the exection result), but it should be easy to detect ("tx ran but failed in checktx/delivertx" along with the log info from the chain)

File: packages/bcp/package.json (hunk 1 above)
Comment (webmaster128) [reply packages/bcp/package.json:37]: Yeah this was never needed here. Will created that for IOV because of CORS issues in the original Gaia docker container.

File: packages/sdk38/src/cosmosclient.searchtx.spec.ts (hunk 4 above)
Comment (ethanfrey) [reply packages/sdk38/src/cosmosclient.searchtx.spec.ts:111]: Do you want to create an issue for this approach and we can discuss this?

I would not change it here, but for sdk39 I think we will change lots of interfaces, so good to check all issues with the current design so we can improve on them,

File: packages/bcp/src/cosmosconnection.spec.ts (hunk 2 above)
Comment (webmaster128) [reply packages/bcp/src/cosmosconnection.spec.ts:57]: A few PRs ago, we decided to use the mnemonic "economy stock theory fatal elder harbor …" for the faucet package only. The usage here was forgotten. With browser testing and the new independence of faucet, bcp, cosmwasm, we get new parallel running conflicts that did not occur in the node testing by pure luck.

**Comment by webmaster128** [comment]
> let's see what this does to CI times.

This is no big deal because we use the same `build` job's artifacts that we create for `test` and only run one fast headless browser.

File: packages/sdk38/src/cosmosclient.searchtx.spec.ts (hunk 4 above)
Comment (webmaster128) [reply packages/sdk38/src/cosmosclient.searchtx.spec.ts:111]: Okay, I put it here: https://github.com/CosmWasm/cosmwasm-js/issues/199
There was https://github.com/CosmWasm/cosmwasm-js/issues/194 already which shows the current solution is unsufficient.
